### PR TITLE
Proposal: auto generate helm chart docs

### PR DESCRIPTION
**Pull request by 0-sv**
# 1
Maybe this approach can use the input of the rest of the developers, but I was working on a local install and I thought the chart is kinda confusing. I want answers to basic questions like what each option does, and preferably I can make up some assumptions based on the `values.yaml` alone. I would propose that we stick more to "standards", if I look at the `prometheus-operator` `values.yaml` for example, then it is more obvious to me what the options are and what I can tune. Just trying to approach our Chart from a blank slate.

# 2
I also put some feature flags on cz and lint because it is so annoying. In a PR workflow the developer can check if the linting is correct anyway at the end. Nice that the rest thinks it is useful, but I don't want it to interrupt my workflow.

### CODE REVIEW

File: .env.sample [hunk 1]
```diff
@@ -0,0 +1,3 @@
+ENABLE_CZ=true
+ENABLE_NPM_LINT=true
+ENABLE_HELM_DOCS=true
```
Comment (Morriz) [inline .env.sample:3]: look, enforcement of style is not optional, and we should all have the same workflow. I am against these flags as it makes devs bypass what needs to be uniformly applied.

If you have suggestions to make that better, great. But don't attempt to undermine that with such flags please.

**Review by Morriz — COMMENTED** [review]
imo this PR is too convoluted, and goes against what we already decided before. I suggest to close it. If you have ideas about documentation or anything related to presentation please consult with the business owners before creating PRs.

File: .husky/pre-commit [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
```
Comment (j-zimnowoda) [inline .husky/pre-commit:1]: I think documentation could be generated on helm release

**Comment by j-zimnowoda** [comment]
@0-sv thanks for your initiative. I can say that a lot of charts provides autogenerate table with possible options, as JSON schemas (even those in yaml format) are difficult to read.
Current approach provides documentation (possible alternatives) inside the `values.yaml` file.
If we could automate generating essential documentation for `cluster`, `otomi`, `oidc`, `dns`, `charts.external-dns`, `charts.cert-manager` settings, then I consider it as a great addition.

@Morriz would you reconsider Sebastians idea?

**Comment by Morriz** [comment]
> @Morriz would you reconsider Sebastians idea?

Sure, but this is not a new idea at all. It's just that this PR was not tailored to the intended outcome.

**Comment by Morriz** [comment]
FYI: this was started in redkubes/otomi-tasks#38

**Pull request closed without merging**